lib/FieldType/Mapper/CountryFormMapper.php

```diff
@@ -8,27 +8,30 @@
  */
 namespace EzSystems\RepositoryForms\FieldType\Mapper;
 
+use eZ\Publish\API\Repository\FieldTypeService;
+use EzSystems\RepositoryForms\Data\Content\FieldData;
 use EzSystems\RepositoryForms\Data\FieldDefinitionData;
-use EzSystems\RepositoryForms\FieldType\DataTransformer\CountryValueTransformer;
+use EzSystems\RepositoryForms\FieldType\DataTransformer\MultipleCountriesValueTransformer;
+use EzSystems\RepositoryForms\FieldType\DataTransformer\SingleCountryValueTransformer;
 use EzSystems\RepositoryForms\FieldType\FieldDefinitionFormMapperInterface;
+use EzSystems\RepositoryForms\FieldType\FieldValueFormMapperInterface;
 use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
 use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
-class CountryFormMapper implements FieldDefinitionFormMapperInterface
+class CountryFormMapper implements FieldDefinitionFormMapperInterface, FieldValueFormMapperInterface
 {
-    /**
-     * @var array Array of countries from ezpublish.fieldType.ezcountry.data
-     */
+    /** @var array Array of countries from ezpublish.fieldType.ezcountry.data */
     protected $countriesInfo;
 
-    /**
-     * @param array $countriesInfo
-     */
-    public function __construct(array $countriesInfo)
+    /** @var FieldTypeService */
+    private $fieldTypeService;
+
+    public function __construct(array $countriesInfo, FieldTypeService $fieldTypeService)
```

Comment:
@webhdx As it is constructor it should be fine, but changes like this still needs to be documented as a BC _change_. [Like in kernel](https://github.com/ezsystems/ezpublish-kernel/blob/master/doc/bc/changes-6.9.md) that would be `doc/bc/changes-1.10.md` in repo here.

Reply:
@bdunogier @andrerom I don't think we can pull it off without doing some BC breaks here and there. It's possible to avoid BC breaks but the drawback would be a lot of duplicated code (duplicating FieldType's `fromHash()` functionality in many cases). This particular case (Country fieldtype) is actually my mistake as it doesn't use `FieldTypeService` at all and I will remove it in a minut. RichText and Relation (single and multiple) also had constructor changed.

Since we are following semver and BC breaks now shouldn't be an option right now, do you think we should bump `repository-forms` to 2.x to ship it with Hybrid UI?

Reply:
@webhdx let me rephrase my reply above: I don't think constructor changes is a BC break in semver sense, it's something we do all the time in kernel and most other repos when we need to fix something and need more dependencies to do it. So just add a bc note about it, it's not strictly needed, but an extra heads up to those who _extend_ our classes. Breaking _usage_, aka breaking api would be something else.

Reply:
Now clear. I added a document describing BC changes in 1.10 (I suppose this is next release number). Thanks.